ajax/oauth2.php

```diff
@@ -59,8 +55,14 @@
 		} else if ($step == 2 && isset($_POST['code'])) {
 			try {
 				$token = $client->authenticate((string)$_POST['code']);
+				if (isset($token['error'])) {
+					OCP\JSON::error(['data' => [
+						'message' => $l->t('Step 2 failed. Exception: %s', [$token['error_description']])
+					]]);
+					return;
+				}
 				OCP\JSON::success(['data' => [
-					'token' => $token
+					'token' => json_encode($token)
 				]]);
```

Comment:
We need to encode the token?
Isn't this will be automatically done by this function `OCP\JSON::success`